ls1intum/Artemis
# `General`: Allow instructors to import users via csv into course groups

### Checklist

### Motivation and Context


Closes #3793. I extended the functionallities, so you can import users to each CourseGroup via CSV 
### Description

I added a new button to the CourseGroupComponent. Pressing the button allows the user to drag in a CSV to import Users to a course

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Management
3. Select a Course
4. Open the students/tutors/editor/instructor group
5. See the new button
6. Check the import dialog for grammer mistakes
7. Check the import functionallity by dragging in a csv in the given format:
![image](https://user-images.githubusercontent.com/65229601/141779858-d1084f71-ec22-4e45-ba48-88bdd199c16e.png)
8.Check, whether the exam-import-function still works as expected (adding student to the exam AND the course) by applying the same steps (6-8) in the students tab of an exam


### Review Progress





- Code Review
- Manual Tests

### Test Coverage




 global coverage: 76,97%
 users-import-dialog.component.ts    87,78%

### Screenshots


<img width="438" alt="button" src="https://user-images.githubusercontent.com/65229601/140644440-60b66990-8641-478b-9b6e-82fbebf05c40.png">
<img width="401" alt="dialog" src="https://user-images.githubusercontent.com/65229601/140644441-c4616e83-a8a4-4506-8a28-0e63859cbacb.png">


the screenshot .csv
[Mappe2.csv](https://github.com/ls1intum/Artemis/files/7538867/Mappe2.csv)
